integrations/pjrt/src/iree_pjrt/common/api_impl.cc

```diff
@@ -852,6 +918,131 @@ iree_status_t DeviceInstance::HostBufferToDeviceZeroDim(
   return iree_ok_status();
 }
 
+iree_status_t DeviceInstance::TransposeBroadcastDeviceBuffer(
+    BufferInstance* buffer, iree_hal_element_type_t element_type,
+    const iree_hal_dim_t* input_dims, const iree_hal_dim_t* output_dims,
+    const int64_t* perms, size_t num_dims,
+    PJRT_HostBufferSemantics host_buffer_semantics,
+    EventInstance** out_done_with_host_buffer_event,
+    BufferInstance** out_buffer) {
+  if (num_dims > 9) {
```

Comment:
OOC why?

Reply:
Added comment with shared constant: currently it uses `std::array`to avoid a`malloc`. Currently its set to `9` but if this turns out to be an issue we can increase the value.